test/user.test.ts

```diff
@@ -0,0 +1,182 @@
+    const { data } = await saleor.user.requestEmailChange({
+      newEmail: `register+${Date.now().toString()}@example.com`,
+      password: TEST_AUTH_PASSWORD,
+      redirectUrl: API_URI,
+    });
+    expect(data?.requestEmailChange?.errors).toHaveLength(0);
+  });
+
+  it("updates the user's first name", async () => {
+    const { data } = await saleor.user.updateAccount({
+      input: {
+        firstName: "",
+      },
+    });
+    expect(data?.accountUpdate?.user?.firstName).toBe("");
+    expect(data?.accountUpdate?.errors).toHaveLength(0);
+  });
+
+  it("creates the user's account address", async () => {
+    const { data } = await saleor.user.createAccountAddress({
+      input: testAddress,
+    });
+
+    if (data?.accountAddressCreate?.user?.addresses?.length) {
+      expect(
+        data?.accountAddressCreate?.user?.addresses[
+          data?.accountAddressCreate?.user?.addresses?.length - 1
+        ].firstName === testAddress.firstName
+      );
+    }
+    expect(data?.accountAddressCreate?.errors).toHaveLength(0);
+  });
+
+  it("updates the user's account address", async () => {
+    const cache = client.readQuery({
+      query: USER,
+    });
+    const index = cache.me.addresses.length - 1;
+    const newTestName = "New test name";
+    const { data } = await saleor.user.updateAccountAddress({
+      id: cache.me.addresses[index].id,
+      input: {
+        ...testAddress,
+        firstName: newTestName,
+      },
+    });
+    const updatedCachedUser = client.readQuery({
+      query: USER,
+    });
+    if (data?.accountAddressUpdate?.user?.addresses?.length) {
+      expect(
+        data?.accountAddressUpdate?.user?.addresses[index]?.firstName
+      ).toBe(newTestName);
+      expect(updatedCachedUser?.me?.addresses[index]?.firstName).toBe(
+        newTestName
+      );
+    }
+    expect(data?.accountAddressUpdate?.errors).toHaveLength(0);
+  });
+
+  it("sets address as a default billing address", async () => {
+    const cache = client.readQuery({
+      query: USER,
+    });
+    const index = cache.me.addresses.length - 1;
+    const addressId = cache.me.addresses[index].id;
+    const oldAddress = cache.me.defaultBillingAddress;
+    const { data } = await saleor.user.setAccountDefaultAddress({
+      id: addressId,
+      type: "BILLING",
+    });
+    expect(
+      data?.accountSetDefaultAddress?.user?.defaultBillingAddress?.id
+    ).toBe(addressId);
+    expect(data?.accountSetDefaultAddress?.errors).toHaveLength(0);
+
+    // Set back default testing address
+    const { data: revertedData } = await saleor.user.setAccountDefaultAddress({
+      id: oldAddress.id,
+      type: "BILLING",
+    });
+    expect(
+      revertedData?.accountSetDefaultAddress?.user?.defaultBillingAddress?.id
+    ).toBe(oldAddress.id);
+    expect(revertedData?.accountSetDefaultAddress?.errors).toHaveLength(0);
+  });
+
+  it("deletes user address", async () => {
```

Comment:
If run on staging with a shared test account, then it's not guaranteed this test will pass - there can be no address.

Reply:
If you want to be sure that some address exists - create new one